### PR TITLE
Retriever doesn't stop on errors 

### LINKED ISSUE #10: Allow individual requests to fail without ending the process
Especially as the retriever grows, it would be quite inefficient to crash the process on any error encountered without good feedback about what has succeeded and what hasn't.

Instead, contain errors, end affected streams, allow other requests to proceed as normal. Report all successes and failures as they happen, and again during the callback once retrieving is done.

First step will be to unhook the errback in `recordCallback` from `wrappedCb`.

### PR DESCRIPTION
**Pull request by wpears**
Rather than exiting after an error is encountered on a record, errors are trapped and recorded. The final callback is now called with an array of all encountered errors and an array of successfully processed records.

Wait for #9 before merging.

Closes #10

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,13 +28,16 @@
...
     "csv-to-vrt": "0.0.2",
+    "end-of-stream": "^1.1.0",
     "fs-extra": "^0.18.4",
-    "hyperquest": "^1.2.0",
-    "s3-upload-stream": "^1.0.7",
-    "unzip": "^0.1.11"
-  },
-  "devDependencies": {
-    "tape": "^4.0.0"
+    "ftp": "^0.3.10",
```
Comment (hkeeler) [inline package.json:35]: Why the move to `devDependencies`?  Other than `tape`, most of these seem like they should go under `dependencies`.

Also, where is `end-of-stream` used?  I see its a dependency of `pump`.  Is this just to get a newer or specific version compared to comes with `pump`?  Or is this somehow necessary for the shrinkwrappings?

**Comment by hkeeler** [comment]
Review complete.  Looks good.  Just the one question about `dependencies` vs. `devDependencies`.

File: package.json (hunk 1 above)
Comment (hkeeler) [inline package.json:35]: Nevermind about where `end-of-stream` is used.  I see its used directly in the next PR.  Still curious about deps vs. devDeps.

Comment (wpears) [inline package.json:35]: I agree re: dev dependencies. In the later PRs I've actually moved everything to plain dependencies and I've stopped shrinkwrapping (which is really a premature optimization imo until development settles down. Having to reshrinkwrap for every new dependency kinda clutters to commits and is another thing to forget). Once I start shrinkwrapping again, I'll set up a pre-commit hook that shrinkwraps automatically so I feel minimum friction.

The reason it's done in the first place is to have exact dependencies so we don't end up with untested broken code due to someone pushed a bad 1.x.x version

**Comment by hkeeler** [comment]
Cool.  Merging.

**Pull request merged**